cmd/heimdalld/cmd/commands.go

```diff
@@ -174,18 +175,53 @@ func initRootCmd(
 		PostSetup: func(svrCtx *server.Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error {
 			helper.InitHeimdallConfig("")
 
-			// wait for the rest server to start
-			resultChan := make(chan string)
-			// TODO take this back to 5 minutes after migration is done
-			timeout := time.After(120 * time.Minute)
+			// wait for the rest server to start.
+			// buffer the channel to prevent race conditions
+			resultChan := make(chan string, 1)
```

Comment:
Hmm wonder what race conditions you encountered ?

Reply:
I removed the comment here 5d69cda1566f92bcb6ff3d036f624bde59cc4354 because it was misleading. Actually, it's more about sync that race. 
The reason of the buffered channel are mainly to avoid any block on checkServerStatus waiting for receivers. This  can potenially stall the poller and (for example) delay the shutdown.
The buf channel can be dropped even if receiver is not ready
Thanks.